content/en/docs/ops/diagnostic-tools/cni/index.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Troubleshooting CNI
+description: Describes tools and techniques to diagnose issues with CNI.
+weight: 90
+keywords: [debug,cni]
+owner: istio/wg-networking-maintainers
+test: n/a
+---
+
+This page describes how to troubleshoot issues with Istio CNI.
+Before reading this, you should read the [CNI installation and operation guide](/docs/setup/additional-setup/cni/).
+
+## Log
+
+The Istio CNI plugin log provides information about how CNI plugin configures application pod traffic redirection
+based on pod spec.
+
+The plugin runs in the container runtime process space, so you can collect CNI plugin log from the `kubelet` log.
+To ease the log collection, the CNI plugin also sends its log to the `istio-cni-node` DaemonSet.
+Due to this, You can get the plugin log simply by getting `istio-cni-node` DaemonSet pod log.
+The default log level for CNI plugin is at `info` level, to get more detailed log from the CNI plugin,
+you can tune plugin log level via install option `values.cni.logLevel` and restart CNI DaemonSet pod.
+
+In addition to CNI plugin log, the Istio CNI DaemonSet pod log also provides information about CNI network plugin installation,
+and [race condition repairing](/docs/setup/additional-setup/cni/#race-condition-mitigation).
+
+## Monitoring
+
+The CNI DaemonSet [generates metrics](/docs/reference/commands/install-cni/#metrics),
+which can be used to monitor CNI installation, readiness, and race condition mitigation.
+Prometheus scraping annotations (`prometheus.io/port`, `prometheus.io/path`) are added to `istio-cni-node` DaemonSet pod by default.
+You can collect the generated metrics via standard Prometheus configuration.
+
+## Diagnose pod start-up failure
```

Comment:
We should have a log of when the istio-validation container fails, and what it means.

Reply:
Done.